src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -812,26 +822,42 @@ void ClearSecretData(uint8_t * buf, uint32_t len)
         }                                                                                                                          \
     } while (0)
 
+typedef struct Spake2p_Context
+{
+    EC_GROUP * curve;
+    BN_CTX * bn_ctx;
+    const EVP_MD * md_info;
+} Spake2p_Context;
+
+Spake2p_P256_SHA256_HKDF_HMAC::Spake2p_P256_SHA256_HKDF_HMAC(void) :
+    Spake2p(kP256_FE_Length, kP256_Point_Length, kSHA256_Hash_Length)
+{
+    memset(mSpake2pContext, 0, sizeof(mSpake2pContext));
+    nlSTATIC_ASSERT_PRINT(sizeof(mSpake2pContext) >= sizeof(Spake2p_Context), "Need more memory for Spake2p Context");
+}
+
 CHIP_ERROR Spake2p_P256_SHA256_HKDF_HMAC::InitInternal(void)
 {
     CHIP_ERROR error  = CHIP_ERROR_INTERNAL;
     int error_openssl = 0;
 
-    context.curve   = NULL;
-    context.bn_ctx  = NULL;
-    context.md_info = NULL;
+    Spake2p_Context * context = reinterpret_cast<Spake2p_Context *>(mSpake2pContext);
```

Comment:
what if you wrote little static inline functions that did the check, the cast, and returned the appropriate type, making it clearer the requirements for the variable, putting all the hacky stuff together?